plugins/modules/net_tools/dnsimple.py

```diff
@@ -73,7 +71,7 @@
     type: 'bool'
     default: no
 requirements:
-  - "dnsimple >= 1.0.0"
+  - "dnsimple >= 2.0.0"
```

Comment:
This is a breaking change. Is it necessary to break compatibility with dnssimple < 2.0.0?

Reply:
Yes, I guess. Because dnsimple < 2 deprecated and no longer available.

Reply:
Technically we could keep this module backwards compatible with dnsimple < 2 with some major changes to the module. However, as @Akasurde  said I don't know if this makes sense / is desirable.

Reply:
https://blog.dnsimple.com/2016/12/api-v2-stable/ says that v1 is discontinued on October 1st, 2017. So I think we should remove it since dnsimple is SaaS.

Reply:
According to the code comments (https://github.com/ansible-collections/community.general/pull/2946/files#diff-5d983a5b601bdc6628f8aa0911b85fe1c3a2c33038a0d35982b5fefce8c3a518L189), dnsimple >= 1.0.0 was using the v2 API as well; only < 1.0.0 was using the v1 API. So dnsimple versions between 1.0.0 and before 2.0.0 seem to be working with the v2 API as well.

Reply:
You are right: https://github.com/dnsimple/dnsimple-python/blob/1.0.0/dnsimple/dnsimple.py#L63
I will try to make this module work with dnsimple verison 1 and 2. Thank you for the review so far!

Reply:
I changed a lot of code in the module to make it compatible with both dnsimple versions: https://github.com/ansible-collections/community.general/pull/2946/commits/3e6096d00fec2056dc06d8c7027c298888d3c370

Reply:
I think it makes sense to deprecate support for dnsimple < 2.0.0; so that we can eventually clean this up :) It's probably best to deprecate it in c.g 4.0.0 and remove it in c.g 5.0.0, so the deprecation should be added in a follow-up PR that isn't backported to stable-3. Then from c.g 3.4.0 on, users can use also dnsimple >= 2.0.0, from c.g 4.0.0 on they get a warning if they're using dnsimple < 2.0.0, and from c.g 5.0.0 on they have to use dnsimple >= 2.0.0. That should provide a fair migration path for users, and a perspective for us to get rid of the extra code :)

Reply:
that sounds reasonable. I moved the code to separate classes for python-dnsimple 1 / 2 which should make the cleanup process much more straightforward